ably.ts.d

```diff
@@ -0,0 +1,400 @@
+
+  subscribe: (eventOrCallback: String | messageCallback, listener?: messageCallback) => void;
+
+  unsubscribe: (eventOrCallback?: String | messageCallback, listener?: messageCallback) => void;
+
+  publish: (messages: any, MessageDataOrCallback?: any | errorCallback, callback?: errorCallback) => void;
+
+}
+
+export declare class Channels {
+  get: (name: String, channelOptions?: ChannelOptions) => RealtimeChannel;
+
+  release: (name: String) => void;
+}
+
+export declare class Message {
+  constructor();
+
+  fromEncoded: (JsonObject: String, channelOptions: ChannelOptions) => Message;
+
+  fromEncodedArray: (JsonArray: String, channelOptions: ChannelOptions) => Array<Message>;
+
+  clientId: String;
+  connectionId: String;
+  data: any;
+  encoding: String;
+  extras: String;
+  id: String;
+  name: String;
+  timestamp: number; // Time
+}
+
+export declare class PresenceMessage {
+  fromEncoded: (JsonObject: any, channelOptions?: ChannelOptions) => PresenceMessage;
+
+  fromEncodedArray: (JsonArray: Array<any>, channelOptions?: ChannelOptions) => Array<PresenceMessage>;
+
+  action: PresenceAction;
+  clientId: String;
+  connectionId: String;
+  data: any;
+  encoding: String;
+  id: String;
+  timestamp: number; // Time
+}
+
+export declare class Rest {
+  constructor(options: String | ClientOptions);
+  auth: Auth;
+  channels: Channels;
+  request: (method: string, path: string, parmas?: any, body?: Array<any> | any, headers?: any, callback?: (Error: ErrorInfo, results: HttpPaginatedResponse) => void) => void;
+  stats: (ParamsOrCallback?: any, callback?: PaginatedResultCallback) => void;
+  time: (paramsOrCallback?: any, callback?: (Error: ErrorInfo, time: number ) => void) => void; // time
+}
+
+export declare class Realtime {
+  constructor(options: String | ClientOptions);
+
+  auth: Auth;
+  channels: Channels;
+  clientId: String;
+  connection: Connection;
+
+  request: (method: string, path: string, parmas?: any, body?: Array<any> | any, headers?: any, callback?: (Error: ErrorInfo, results: HttpPaginatedResponse) => void) => void;
+
+  stats: (ParamsOrCallback?: any, callback?: PaginatedResultCallback) => void;
+
+  close: () => void;
+
+  connect: () => void;
+
+  time: (paramsOrCallback?: any, callback?: (Error: ErrorInfo, time: number ) => void) => void; // time
+
+}
+
+export declare class Connection extends EventEmitter {
+  errorReason: ErrorInfo;
+  id: String;
+  key: String;
+  recoveryKey: String;
+  serial: number;
+  state: ConnectionState;
+
+  close: () => void;
+
+  connect: () => void;
+
+  ping: (callback?: (Error: ErrorInfo, ResponseTime: number ) => void ) => void;
+}
+
+export declare class Stats {
+  all: StatsMessageTypes;
+  apiRequests: StatsRequestCount;
+  channels: StatsResourceCount;
+  connections: StatsConnectionTypes;
+  inbound: StatsMessageTraffic;
+  intervalId: String;
```

Comment:
Well it doesn't actually come from the Ruby lib btw. It comes from the Java lib, which I copied in Ruby, and simply reference the Ruby in our docs because of rubydocs which makes it easier to browse the definitions.  Anyway, @SimonWoolf can we add an issue to add that support in the JS lib down the line because it should match the IDL.